### PR TITLE
[MRG] Refactor read_network_configuration for GUI

### PR DESCRIPTION
**Pull request by gtdang**
Changes were made to the read_network_configuration function that reads hierarchical json and creates a Network object. These changes were made for upcoming GUI update to support the new json format. 

1. Refactored dictionary to Network conversion sections to its own function. 
2. Added an argument to not read external biases
3. Updated and cleaned-up test asset files 

split from #808

### CODE REVIEW

**Comment by gtdang** [comment]
@ntolley This one is ready for merge.

File: hnn_core/hnn_io.py [hunk 1]
```diff
@@ -381,31 +381,27 @@ def _order_drives(gid_ranges, external_drives):
     return ordered_drives
 
 
-@fill_doc
-def read_network_configuration(fname, read_drives=True):
-    """Read network from a json configuration file.
+def dict_to_network(net_data,
+                    read_drives=True,
+                    read_external_biases=True):
+    """ Converts a dict of network configurations to a Network
```
Comment (ntolley) [inline hnn_core/hnn_io.py:387]:
```suggestion
    """Converts a dict of network configurations to a Network
```

File: hnn_core/hnn_io.py [hunk 2]
```diff
@@ -454,3 +451,35 @@ def read_network_configuration(fname, read_drives=True):
         net.clear_drives()
 
     return net
+
+
+def read_network_configuration(fname,
+                               read_drives=True,
+                               read_external_biases=True):
+    """ Read network from a json configuration file.
```
Comment (ntolley) [inline hnn_core/hnn_io.py:459]:
```suggestion
    """Read network from a json configuration file.
```

**Review by ntolley — APPROVED** [review]
Just some small nitpicks otherwise happy to merge!

**Comment by gtdang** [comment]
> Just some small nitpicks otherwise happy to merge!
Thanks, committed those changes. Should be ready to merge.

**Pull request merged**